src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3221,6 +3221,10 @@ tx_gateway_frontend::get_all_transactions() {
           model::tx_manager_nt.ns, model::tx_manager_nt.tp, pa.id);
         auto ntp_res = co_await get_all_transactions_for_one_tx_partition(
```

Comment:
can we skip listing transactions if the node is not coordinator ?

Reply:
Protocol wise, yes, it's ok to skip partition if the node isn't a coordinator.

Should we check leadership before invoking `get_all_transactions_for_one_tx_partition`? Probably not, the metadata cache & the leadership table may be out of date and asking a consensus requires jumping to the right core what `get_all_transactions_for_one_tx_partition` is doing anyway so in this case optimistic approach works fine.